### PR TITLE
Retry logic, criteria getAll tests added

### PR DESCRIPTION
**Pull request by Anarios**

### CODE REVIEW

File: dist/src/common.class.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /**
  * plain object for arbitrary data
  */
-export declare type Data = {
+export interface Data {
```
Comment (MarkHerhold) [inline dist/src/common.class.d.ts]: Why the change?

Comment (Anarios) [reply dist/src/common.class.d.ts]: Linting fails otherwise. We can keep the changes, or alter a linting rule.

File: src/http.ts [hunk 2]
```diff
@@ -76,6 +78,46 @@ export class BadgeUpHttp {
...
+            } else {
+                reject(new Error(response.statusText));
+            }
+        }
+        function failure(error) {
+            limit--;
+            if (limit >= 0) {
+                return fetchUrl();
+            } else {
+                reject(error);
```
Comment (MarkHerhold) [inline src/http.ts]: I believe this implementation is flawed because we can end up calling `reject()` on the wrapping promise multiple times.
